openedx/core/djangoapps/course_views/course_views.py

```diff
@@ -0,0 +1,178 @@
+"""
+Tabs for courseware.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+from xmodule.tabs import CourseTab, link_reverse_func
+
+_ = lambda text: text
+
+
+# Stevedore extension point namespaces
+COURSE_VIEW_TYPE_NAMESPACE = 'openedx.course_view_type'
+
+
+class CourseViewType(object):
+    """
+    Base class of all course view type plugins.
+    """
+    name = None    # The name of the view type, which is used for persistence and view type lookup
+    title = None    # The title of the view, which should be internationalized
+    priority = None    # The relative priority of this view that affects the ordering (lower numbers shown first)
+    view_name = None    # The name of the Django view to show this view
+    tab_id = None    # The id to be used to show a tab for this view
+    is_movable = True    # True if this course view can be moved
+    is_dynamic = False    # True if this course view is dynamically added to the list of tabs
+    is_default = True    # True if this course view is a default for the course (when enabled)
+    is_hideable = False    # True if this course view's visibility can be toggled by the author
+
+    @classmethod
+    def is_enabled(cls, course, django_settings, user=None):  # pylint: disable=unused-argument
```

Comment:
We no longer need to pass in `django_settings` because the tabs have been moved and can import django settings if they need them.